### PR TITLE
Add support for AWS temporary credentials for S3 source and sink

### PR DESCRIPTION
**Pull request by ssmith-google**

### CODE REVIEW

File: widgets/S3-batchsource.json [hunk 1]
```diff
@@ -568,7 +579,11 @@
         {
           "type": "property",
           "name": "accessKey"
-        }
+        },
+	{
+	  "type": "property",
```
Comment (seanzhougoogle) [inline widgets/S3-batchsource.json]: nit: fix the indentation

File: widgets/S3-batchsource.json [hunk 2]
```diff
@@ -548,7 +556,10 @@
         },
         {
           "name": "accessKey"
-        }
+        },
+	{
```
Comment (seanzhougoogle) [inline widgets/S3-batchsource.json]: nit: fix the indentation

File: widgets/S3-batchsink.json [hunk 3]
```diff
@@ -160,7 +168,10 @@
         },
         {
           "name": "accessKey"
-        }
+        },
+	{
```
Comment (seanzhougoogle) [inline widgets/S3-batchsink.json]: nit: fix the indentation

File: src/main/java/io/cdap/plugin/aws/s3/source/S3BatchSource.java [hunk 4]
```diff
@@ -155,6 +164,11 @@ public void validate(FailureCollector collector) {
       if (!containsMacro("path") && (!path.startsWith("s3a://") && !path.startsWith("s3n://"))) {
         collector.addFailure("Path must start with s3a:// or s3n://.", null).withConfigProperty(NAME_PATH);
       }
+      if (!containsMacro("path") && path.startsWith("s3n://") && connection.getSessionToken() != null &&
+            !connection.getSessionToken().isEmpty()) {
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/plugin/aws/s3/source/S3BatchSource.java]: can leverage `!Strings.isNullOrEmpty(connection.getSessionToken())`

File: src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java [hunk 5]
```diff
@@ -171,6 +178,11 @@ public void validate(FailureCollector collector) {
         collector.addFailure("Path must start with s3a:// or s3n://.", null).withConfigProperty(NAME_PATH);
       }
 
+      if (!containsMacro(NAME_PATH) && path.startsWith("s3n://") && sessionToken != null && !sessionToken.isEmpty()) {
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java]: can leverage !Strings.isNullOrEmpty(connection.getSessionToken())

File: docs/S3-batchsink.md [hunk 6]
```diff
@@ -32,6 +32,8 @@ IAM can only be used if the plugin is run in an AWS environment, such as on EMR.
 
 **Access Key:** Amazon access key required for authentication.
 
+**Session Token:** Amazon session token required for authentication. Only required for temporary credentials.
```
Comment (seanzhougoogle) [inline docs/S3-batchsink.md:35]: would better mention only support s3n schema here

File: docs/S3-batchsource.md [hunk 7]
```diff
@@ -33,6 +33,8 @@ IAM can only be used if the plugin is run in an AWS environment, such as on EMR.
 
 **Access Key:** Amazon access key required for authentication.
 
+**Session Token:** Amazon session token required for authentication. Only required for temporary credentials.
```
Comment (seanzhougoogle) [inline docs/S3-batchsource.md:36]: would better mention only support s3n schema here

File: docs/S3-connector.md [hunk 8]
```diff
@@ -18,10 +18,12 @@ IAM can only be used if the plugin is run in an AWS environment, such as on EMR.
 
 **Access Key:** Amazon access key required for authentication.
 
+**Session Token:** Amazon session token required for authentication. Only required for temporary credentials.
```
Comment (seanzhougoogle) [inline docs/S3-connector.md:21]: would better mention only support s3n schema here

**Review by seanzhougoogle — APPROVED** [review]
LGTM with some minor comments

File: docs/S3-batchsink.md (hunk 6 above)
Comment (ssmith-google) [reply docs/S3-batchsink.md:35]: Done.

File: docs/S3-batchsource.md (hunk 7 above)
Comment (ssmith-google) [reply docs/S3-batchsource.md:36]: Done

File: docs/S3-connector.md (hunk 8 above)
Comment (ssmith-google) [reply docs/S3-connector.md:21]: Done.

File: src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java (hunk 5 above)
Comment (ssmith-google) [reply src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java]: Done.

File: src/main/java/io/cdap/plugin/aws/s3/source/S3BatchSource.java (hunk 4 above)
Comment (ssmith-google) [reply src/main/java/io/cdap/plugin/aws/s3/source/S3BatchSource.java]: Done.

File: widgets/S3-batchsink.json (hunk 3 above)
Comment (ssmith-google) [reply widgets/S3-batchsink.json]: Done.

File: widgets/S3-batchsource.json (hunk 2 above)
Comment (ssmith-google) [reply widgets/S3-batchsource.json]: Done.

File: widgets/S3-batchsource.json (hunk 1 above)
Comment (ssmith-google) [reply widgets/S3-batchsource.json]: Done.

**Pull request merged**